pkg/quota/v1/resources.go

```diff
@@ -211,6 +212,18 @@ func Intersection(a []corev1.ResourceName, b []corev1.ResourceName) []corev1.Res
 	return result
 }
 
+// Difference returns the list of resources resulting from a-b
+func Difference(a []corev1.ResourceName, b []corev1.ResourceName) []corev1.ResourceName {
```

Comment:
This is a fairly expensive function - three map allocations and then the resource array.  Even though it's the error case this could be cheaper.

Reply:
rewrote with a single alloc (and rewrote `Intersection` to do a single alloc, and `Contains` to do zero allocs), and added tests